mesonbuild/interpreterbase/_unholder.py

```diff
@@ -5,14 +5,16 @@
 
 import typing as T
 
-from .baseobjects import InterpreterObject, MesonInterpreterObject, ObjectHolder, HoldableTypes
+from .baseobjects import InterpreterObject, MesonInterpreterObject, ObjectHolder, HoldableTypes, MutableInterpreterObject
 from .exceptions import InvalidArguments
 from ..mesonlib import HoldableObject, MesonBugException
 
 if T.TYPE_CHECKING:
     from .baseobjects import TYPE_var
 
 def _unholder(obj: InterpreterObject) -> TYPE_var:
+    if isinstance(obj, MutableInterpreterObject):
+        obj.mark_used()
```

Comment:
This is rather excessive. Before this commit we set `obj.used = True` directly in func_configure_file() and also in the cmake module. Which this commit doesn't clean up either. We cannot be more restrictive than that except via a warning...

In retrospect, the fact that this added new ways to mark_used() but didn't delete the old ways was a bit of a red flag that the logic wasn't equivalent.